@damassi => Auction results container

mzikherman opened the pull request:
![bp](https://user-images.githubusercontent.com/1457859/41566466-8e4225b0-7329-11e8-973b-5b309d7fb925.gif)


This renders a list of auction results, includes pagination bar w/ prev/next, as well as sorting!

Will make some inline comments.

mzikherman commented on src/Styleguide/Components/Pagination.tsx:
```diff
@@ -80,7 +129,7 @@ const PageSpan = ({ mx }) => {
   return (
     <Sans size="3" weight="medium" display="inline" mx={0.5}>
-      <a href="#" className="noUnderline">
+      <a onClick={() => props.onClick()} className="noUnderline">
```
I noticed `href='#'` was causing storybooks to register the click and navigate you in a weird way.

In the real world, this will probably update some URL (and perhaps other stuff), but that shouldn't be in this component right? That should be a callback controlled by the parent?

I wonder, for SEO purposes, we probably _do_ want to be able to render valid `href` tags as well (ones that if you load them directly, will resolve to the right page/sort/filter/etc.). That can all come later I guess?

damassi replied:
We'll need to think this through a bit, but ideally the links update the URL bar with a new query parameter that then updates the filter. And yup, we can do that in another pass.

mzikherman replied:
Cool, yea that can be a further pass since it for sure involves some thinking through.

alloy replied:
My immediate thought is that such things as interacting with the host of the app (the browser in this case) should happen at the root component level.

alloy commented on src/Styleguide/Containers/Artist/AuctionResults/AuctionResults.tsx:
```diff
@@ -0,0 +1,236 @@
+        <Pagination
+          first={this.props.artist.auctionResults.pageCursors.first}
+          last={this.props.artist.auctionResults.pageCursors.last}
+          around={this.props.artist.auctionResults.pageCursors.around}
```
These 3 lines can be replaced with: `{...this.props.artist.auctionResults.pageCursors}`

mzikherman replied:
oh yea cool! i finally got comfortable/started using `const { var, ...notVars } = options` !

damassi commented on src/Styleguide/Components/AuctionResultItem.tsx:
```diff
@@ -14,19 +14,14 @@ const ScaledArtworkImage = props => (
   title: string
   date: string
-  dimensions: Dimensions
+  dimensions: string
```
Any reason the `Dimensions` interface was removed?

mzikherman replied:
This was actually after a discussion with @joeyAghion believe it or not!

Basically, in Gravity we have some code that takes our parsed version of auction lot data (in centimeters), and _also_ converts it into inches with fractional parts, (using `Math::Fractional` in Ruby), like `23 1/2`.

Since we're now going directly to Diffusion from Metaphysics, we still have access to the raw centimeter values, but we're losing this non-trivial fractional display/parsing.

However, since even our raw centimeter parsing can sometimes be inaccurate (and sometimes the inches conversion results in values like `23 1/10`), we decided in our new version maybe we should _only_ show the dimensions (as best we could identify), _exactly as they appear in the source document_.

Diffusion offers that, as `dimension_text`, as well as the centimeter values. So this PR/Relay query winds up just querying and rendering that field, vs any converted values.

If we do want to reinstate some sort of friendly fractional display, we can investigate possible JS solutions in Metaphysics (since we'll have raw centimeters), or we can port that from Gravity to Diffusion.

damassi replied:
Makes sense 👍

damassi commented on src/Styleguide/Components/AuctionResultItem.tsx:
```diff
@@ -77,7 +71,9 @@ export const ExtraSmallAuctionResultItem = (props: AuctionResultItemProps) => {
-            <Serif size="2">Sale: {props.salePrice}</Serif>
+            <Serif size="2">
+              {props.salePrice && `Sale: ${props.salePrice}`}
+            </Serif>
```
This can be hoisted up a bit so that an empty dom node doesn't render:

```jsx
{props.salePrice &&
  <Serif size="2">
     ...
  </Serif>
}
```

mzikherman replied:
Good call.

damassi commented on src/Styleguide/Components/AuctionResultItem.tsx:
```diff
@@ -122,7 +117,7 @@ export const SmallAuctionResultItem = (props: AuctionResultItemProps) => {
 
       <div>
-        <Serif size="2">Sale: {props.salePrice}</Serif>
+        <Serif size="2">{props.salePrice && `Sale: ${props.salePrice}`}</Serif>
```
Same here

damassi replied:
(And all following instances)

damassi commented on src/Styleguide/Components/__stories__/Pagination.story.tsx:
```diff
@@ -4,16 +4,43 @@ import { storiesOf } from "storybook/storiesOf"
+    cursor: "blah",
+    isCurrent: false,
+  }
+
```
Could also do 

```jsx
const cursor = { around, first, last } 

return (
  <React.Fragment>
    <Section title="Responsive">	
      <Pagination {...cursor} />
    </Section>
    <Section title="Large Pagination">	
      <LargePagination {...cursor} />
    </Section>
    ...
  </React.Fragment>
)
```

```

damassi commented on src/Styleguide/Containers/Artist/AuctionResults/AuctionResults.tsx:
```diff
@@ -0,0 +1,236 @@
+      this.loadBefore(startCursor)
+    }
+  }
+  loadNext = () => {
```
Can you add line breaks after each method for readability?

damassi commented on src/Styleguide/Containers/Artist/AuctionResults/index.tsx:
```diff
@@ -0,0 +1,53 @@
+  artistID: string
+}
+
+class ArtistFilter extends React.Component<Props> {
```
This filename and named export can be a little more descriptive and should match 1:1 unless additional exports from this file are needed (in which the filename can be more general):

```jsx
export const AuctionResultsFilter = ContextConsumer(props => {
  const { artistID, relayEnvironment } = props

  return (
    <Subscribe to={[FilterState]}>
      ...
    </Subscribe>
  )
})
```

damassi replied:
It also seems like `/Containers` should exist within our namespace as opposed to putting everything top-level -- https://github.com/artsy/reaction/tree/master/src/Styleguide/Pages/Artist.

Later I think we can refactor this organization further, since the container will essentially be root of the tab / page (but that kind of cleanup can come in a few rounds)

damassi commented on src/Styleguide/Elements/Select.tsx:
```diff
@@ -5,30 +5,45 @@ import { Sans } from "@artsy/palette"
+      // tslint:disable-next-line:no-console
+      console.log(`Selected ${value}`)
+    },
+  }
```
Can eliminate `defaultProps` for now

damassi commented on src/Styleguide/Elements/Select.tsx:
```diff
@@ -5,30 +5,45 @@ import { Sans } from "@artsy/palette"
-        <option value="2">Artwork year (desc.)</option>
-        <option value="3">Artwork year (asc.)</option>
+    <LargeSelectContainer {...props} p={1}>
+      <select onChange={value => props.onSelect(value.target.value)}>
```
`value` should be `event`

damassi commented on src/Styleguide/Elements/Select.tsx:
```diff
@@ -42,11 +57,12 @@ export const SmallSelect = props => {
-          <option value="1">Recently added </option>
-          <option value="2">Artwork year (desc.)</option>
-          <option value="3">Artwork year (asc.)</option>
+        <select onChange={value => props.onSelect(value.target.value)}>
```
Also here

damassi commented on src/Styleguide/Components/Pagination.tsx:
```diff
@@ -6,39 +6,88 @@ import { Responsive } from "Styleguide/Utils/Responsive"
+      // tslint:disable-next-line:no-console
+      console.log("Clicked onPrev")
+    },
+  }
```
We should lean on VSCode / TypeScript to warn you about necessary things rather than possibly masking through default props. And not necessarily pass anything (logs are kind of noisy unless they're necessary for legit logging, but either way!),  just complete the contract -- could be something like `x => x`, etc.

mzikherman commented on src/Styleguide/Elements/__stories__/Select.story.tsx:
```diff
@@ -4,16 +4,25 @@ import { Select, LargeSelect, SmallSelect } from "../Select"
+    { value: "bark", text: "Dog" },
+    { value: "moo", text: "Cow" },
+  ]
+
```
Oh, yea, nice.

mzikherman commented:
Ok, i think this is mergeable. Moved the `Containers` folder to be under `Artist`, and addressed the PR feedback. It def needs another stylistic pass, but that can be later. Onward to the next one!

damassi commented on src/Styleguide/Pages/Artist/Overview.js:
```diff
@@ -123,7 +123,7 @@ export const Overview = () => {
                   >
                     <Flex width="100%" pb={2} justifyContent="flex-end">
-                      <Select />
+                      <Select options={[{ value: "percy", text: "Cat" }]} />
```
🐈

damassi commented:
Looking good! Mergin' 👍
